### ISSUE TITLE
Still Getting import/no-unresolved ESLint Errors Despite Using Plugin

### ISSUE THREAD

**Issue by machineghost** [tleunen/eslint-import-resolver-babel-module#36]
I have a `.babelrc` that uses the `module-resolver` plugin:

    "plugins": [
      ["module-resolver", {
        "root": ["./frontend/src"],
        "alias": {
          "components": "./frontend/src/components"
        }
      }]
    ]

I also have an `.eslintrc` that, *in theory*, is configured to use Babel to resolve imports via `eslint-import-resolver-babel-module`:

    "parser": "babel-eslint",
    "plugins": ["babel"],
    "settings": {
      "import/resolver": {
        "babel-module": {}
      }
    }

I know the Babel config is working correctly, because when I run Babel it correctly converts things like:

    import 'components/ui/DatetimePicker'

to:

    var _index = require('../../../../frontend/src/components/ui/DatetimePicker/index');

However in my editor, or if I run ESlint directly, a line like:

    import DatetimePicker from 'components/ui/DatetimePicker';

gives me an error:

    5:28  error  Unable to resolve path to module 'components/ui/DatetimePicker'  import/no-unresolved

Any help would be appreciated.

**Comment by andykenward** [comment]
@machineghost try changing the alias to

```
"alias": {
      "components": "./components"
    }
```

as you have set the root to 
```
 "root": ["./frontend/src"],
```
already

**Comment by tleunen** [comment]
Interesting, it should definitely work with your initial config.
Would you have a repo where this happen so it's easier for me to reproduce and figure out what happen in your case?

**Comment by machineghost** [comment]
@andykenward thanks for the suggestion, but it didn't help :(

@tleunen Unfortunately this is a private project and the customer is super-protective, so I can't show the repo, but I'll try to provide everything relevant.  

My entire .babelrc is:

    {
      "presets": [
        "es2015",
        "react"
      ],
      "plugins": [
        ["module-resolver", {
          "root": ["./frontend/src"],
          "alias": {
            "components": "./components"
          }
        }],
        ["transform-decorators-legacy"]
      ]
    }

My .eslitntrc is:

    {
      "extends": "airbnb",
      "env": {
        "browser": true,
        "node": true
      },
      "globals": {
        "NODE_ENV": true,
        "isProduction": true,
        "__ROUTE__": true
      },
      "parser": "babel-eslint",
      "plugins": [
        "babel"
      ],
      "rules": {
        "react/jsx-filename-extension": [
          "error",
          {
            "extensions": [
              ".js",
              ".jsx"
            ]
          }
        ],
        "comma-dangle": [
          "error",
          {
            "arrays": "always-multiline",
            "objects": "always-multiline",
            "imports": "always-multiline",
            "exports": "always-multiline",
            "functions": "ignore"
          }
        ],
        "react/prop-types": 0
      },
      "settings": {
        "import/resolver": {
          "babel-module": {}
        }
      }
    }

Both files are located in my root folder, and then I have (for instance) a file `frontend/src/routes/index.js`, which has:

    import App from 'components/App';

I have a file `frontend/src/components/App.jsx`, but even though I do I still get:

> :17  error  Unable to resolve path to module 'components/App'   import/no-unresolved
  3:17  error  Missing file extension for "components/App"              import/extensions

**Comment by machineghost** [comment]
If there is any other information I can provide please let me know and I will be happy to.

**Comment by tleunen** [comment]
I've been busy with a few personal issues recently so I couldn't spend time on the few open tickets I have in my OSS projects.

But if you could create a repo/project with the code you had in the previous message, it will definitely help (me or others) to find the issue and hopefully a fix since we won't have to setup the initial project ourselves.

**Comment by craigcosmo** [comment]
I also have this problem, And I created a simple demo for you.

https://github.com/craigcosmo/eslint-resolve-plugin-bug-demo

You can `npm run lint` to reproduce

**Comment by tleunen** [comment]
Hey @craigcosmo,

In your case, the reason why it doesn't work in because you're using a glob setting for the root configuration. Currently, the eslint plugin doesn't support that. I need to release a new version of the babel plugin, then make a fix here to make it work properly.

I'll do it later today or tomorrow.

**Comment by craigcosmo** [comment]
Thanks for making such a good babel plugin. It's something I wanted to build when I got time, you made it first :)

**Comment by craigcosmo** [comment]
@tleunen don't know why but problem still persists after update v2.5 and v2.2.1

**Comment by tleunen** [comment]
Yep, this is because I still need to release a new version for the eslint plugin :)
I'm fixing another small issue with custom extensions and I'll release a new version.

**Comment by tleunen** [comment]
Released a 3.0.0 with a fix for the glob.
